src/plugin/sdl/sdl.c

```diff
@@ -693,16 +696,21 @@ static void unlock_surface(void)
 #if THREADED_REND
 static void *render_thread(void *arg)
 {
+  int ret;
   if (thread_ctx) {
-    int ret = SDL_GL_MakeCurrent(window, thread_ctx);
+    ret = SDL_GL_MakeCurrent(window, thread_ctx);
     assert(!ret);
   }
-  while (1) {
+  while (keep_running) {
```

Comment:
This requires mutex.
I would recommend another approach:
in render_mode_lock() you disable
thread cancelability, so that it can be
cancelled only on a semaphore.
Then you do pthread_cleanup_push()
to add your clean-up code.

Reply:
No, it shouldn't. The semaphore is enough. The important thing is to set it to 0 before signalling the thread to wake up.

Reply:
Ok, yes, thanks.